test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertFrom-Json.Tests.ps1

```diff
@@ -106,6 +110,27 @@ Describe 'ConvertFrom-Json Unit Tests' -tags "CI" {
         { $nestedJson | ConvertFrom-Json -AsHashtable:$AsHashtable } |
             Should -Throw -ErrorId "System.ArgumentException,Microsoft.PowerShell.Commands.ConvertFromJsonCommand"
     }
+
+    It 'Can correctly round trip arrays with NoEnumerate switch set to <NoEnumerate>' -TestCases $testCasesWithAndWithoutNoEnumerateSwitch {
+        Param($NoEnumerate)
+        '[ 1, 2 ]' | ConvertFrom-Json -NoEnumerate:$NoEnumerate | ConvertTo-Json | Should -BeLike '`[*1,*2*]'
+    }
+
+    It 'Unravels array elements when NoEnumerate switch is not set' {
+        ('[ 1, 2 ]' | ConvertFrom-Json | Measure-Object).Count | Should -Be 2
+    }
+
+    It 'Sends a Json array as a single element when NoEnumerate switch is set' {
+        ('[ 1, 2 ]' | ConvertFrom-Json -NoEnumerate | Measure-Object).Count | Should -Be 1
+    }
+
+    It 'Cannot round trip single element arrays without NoEnumerate switch' {
+        '[ 1 ]' | ConvertFrom-Json | ConvertTo-Json | Should -Be '1'
+    }
+
+    It 'Can round trip single element arrays with NoEnumerate switch' {
+        '[ 1 ]' | ConvertFrom-Json -NoEnumerate | ConvertTo-Json | Should -BeLike '`[*1*]'
```

Comment:
Instead of `ConvertTo-Json | Should -BeLike '`[*1*]', I suggest `ConvertTo-Json -Compress | Should -Be '[1]'